packages/ui/src/components/editor/panels/Hierarchy/node/index.tsx

```diff
@@ -208,9 +189,9 @@ export const HierarchyTreeNode = (props: HierarchyTreeNodeProps) => {
         console.warn('parent is not defined')
       }
 
-    return (item: any, monitor): void => {
+    return (item: FileDataType | DnDFileType | DragItemType, monitor: DropTargetMonitor): void => {
       if (parentNode) {
-        if (item.files) {
+        if ('files' in item) {
```

Comment:
can we use something other than a static string here?

Reply:
This is also the key in the object. This helps determining the type of the object if those keys are present.
The other way around is to do `(item as any).files`which we probably want to avoid.

Reply:
Maybe checking for the existence of the property in the object is batter syntax wise

Using only a static string just doesn't feel right

Reply:
Actually, [enums](https://www.typescriptlang.org/docs/handbook/enums.html) are used for an alternative to exhaustive constants in JS. Enums are best useful when you have [a finite set of choices](https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html).

[JS recommends to use string](https://arc.net/l/quote/smujllqd) to check within objects.

Reply:
Just confirming that strings are fine here.